### PR TITLE
chore: add create group and add participant structured logs - WPB-6231

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6231" title="WPB-6231" target="_blank">TaskWPB-6231</a>  [iOS] Improve logging
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Refactor logs to have structured logs displayed on datadog. Starting with two flows:
- create group (MLS part)
- add participants (MLS part)
Have a single logging interface with WireLogger.

### Solutions

- Update `Flow` structure
- Add `AggregatedLogger` to have two loggers at the same time via WireLogger (datadog + systemlog)
- Remove tag of `WireLogger` from message (it can be displayed directly in DD)
- Replace Logging.mls by WireLogger.mls
- Add LegacyLogger to port WireLogger to Filesystem

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-mono.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "pins": [
       {
         "package": "DifferenceKit",
-        "repositoryURL": "https://github.com/wireapp/DifferenceKit/",
+        "repositoryURL": "https://github.com/wireapp/DifferenceKit",
```
Comment (netbe) [inline wire-ios-mono.xcworkspace/xcshareddata/swiftpm/Package.resolved]: this is annoying but I don't even know which one is correct?

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift [hunk 2]
```diff
@@ -36,7 +36,7 @@ public final class CreateGroupConversationAction: EntityAction {
         case accessDenied
         case unreachableDomains(Set<String>)
         case nonFederatingDomains(Set<String>)
-        case proccessingError
+        case proccessingError(String?)
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift:39]: needed to keep track of underlying error / messages

File: wire-ios-mono.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (caldrian) [reply wire-ios-mono.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Are you working with Xcode 15.0.1?

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationParticipantsService.swift [hunk 3]
```diff
@@ -150,11 +155,14 @@ public class ConversationParticipantsService: ConversationParticipantsServiceInt
...
                 // Operation was aborted because some users didn't have key packages
                 // We filter them out and retry once
-
+                Flow.addParticipants.checkpoint(description: "retrying failedUsers begin")
                 try await internalAddParticipants(
                     Array(users.subtracting(failedUsers)),
                     to: conversation
                 )
+                // TODO: maybe we have an issueif this fails again
+                // system message is not inserted for first failedUsers!
```
Comment (markusfassbender) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationParticipantsService.swift]: Does it make sense to create a Jira ticket for this `TODO`?

File: wire-ios-system/Source/Logging/Flow.swift [hunk 4]
```diff
@@ -105,8 +105,10 @@ struct FlowLog: LogConvertible, Encodable {
   let event: Event
 
   var logDescription: String {
-    guard
-      let data = try? JSONEncoder().encode(self),
+      let encoder = JSONEncoder()
+      encoder.outputFormatting = .sortedKeys
```
Comment (markusfassbender) [inline wire-ios-system/Source/Logging/Flow.swift:109]: 👍

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift [hunk 5]
```diff
@@ -299,7 +299,9 @@ extension ConversationCreationController: AddParticipantsConversationCreationDel
                 enableReceipts: values.enableReceipts,
                 messageProtocol: values.encryptionProtocol == .proteus ? .proteus : .mls
             ) { [weak self] in
-                guard let self = self else { return }
+                guard let self = self else {
+                    assertionFailure("expect ConversationService not to be <nil>")
+                    return }
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift]: 💅 the indentation looks off. 
Is the message "expect ConversationService not to be <nil>" correct? I think it's about self (`ConversationCreationController`) here?

**Review by markusfassbender — APPROVED** [review]

File: wire-ios-mono.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (netbe) [reply wire-ios-mono.xcworkspace/xcshareddata/swiftpm/Package.resolved]: yes

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationParticipantsService.swift (hunk 3 above)
Comment (netbe) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationParticipantsService.swift]: it's just something I was wondering reading the code, do you have any opinion on it?

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift (hunk 5 above)
Comment (netbe) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift]: fixed

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift [hunk 6]
```diff
@@ -247,6 +263,7 @@ public final class ConversationService: ConversationServiceInterface {
             unqualifiedUserIDs = usersExcludingSelfUser.compactMap(\.remoteIdentifier)
         }
 
+        createGroupFlow.checkpoint(description: "sending request to backend start")
```
Comment (typfel) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift]: Don't we log all networks request? Isn't it a bit redundant to log this as well.

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift [hunk 7]
```diff
@@ -265,6 +282,7 @@ public final class ConversationService: ConversationServiceInterface {
             self.context.perform {
                 switch result {
                 case .success(let objectID):
+                    self.createGroupFlow.checkpoint(description: "sending request to backend success")
```
Comment (typfel) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift]: Also redundant?

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift (hunk 6 above)
Comment (netbe) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift]: yes we do, but when filtering on this flow, the network logs will not show.
I wanted to know when it started. Maybe there's a way to show it without this checkpoint

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift (hunk 7 above)
Comment (netbe) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift]: removed

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift (hunk 6 above)
Comment (netbe) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift]: removed for now

**Review by typfel — APPROVED** [review]

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationParticipantsService.swift (hunk 3 above)
Comment (netbe) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationParticipantsService.swift]: solved checked

**Pull request merged**